lib/cfa/augeas_parser.rb

```diff
@@ -89,12 +131,20 @@ def []=(key, value)
       end
     end
 
+    # @param matcher [Matcher]
+    # @return [Array<AugeasElement>] matching elements
+    # @todo for consistency this should allow &matcher too
```

Comment:
yes, with new matcher that can be constructed from block, it is possible. Make sense.